src/js/components/Text/Text.js

```diff
@@ -24,19 +26,22 @@ const Text = forwardRef(
   ) => {
     const textRef = useForwardedRef(ref);
     const [textTruncated, setTextTruncated] = useState(false);
-
-    // place the text content in a tip if truncate === 'tip'
-    // and the text has been truncated
-    useEffect(() => {
-      if (truncate === 'tip') {
-        if (
-          textRef.current &&
-          textRef.current.scrollWidth > textRef.current.offsetWidth
-        ) {
-          setTextTruncated(true);
-        }
+    const updateTip = () => {
+      setTextTruncated(false);
+      if (
+        truncate === 'tip' &&
+        textRef.current &&
+        textRef.current.scrollWidth > textRef.current.offsetWidth
+      ) {
+        setTextTruncated(true);
       }
-    }, [children, textRef, truncate]);
+    };
+
+    useLayoutEffect(() => {
+      window.addEventListener('resize', updateTip);
+      updateTip();
+      return () => window.removeEventListener('resize', updateTip);
+    }, []);
```

Comment:
```suggestion
    }, [updateTip]);
```

Reply:
Thank you for your suggestion.
I move the updateTip inside to useLayoutEffect().